test/ocsp/helper/helper.go

```diff
@@ -16,12 +16,70 @@ import (
 	"golang.org/x/crypto/ocsp"
 )
 
-var method = flag.String("method", "GET", "Method to use for fetching OCSP")
-var urlOverride = flag.String("url", "", "URL of OCSP responder to override")
-var hostOverride = flag.String("host", "", "Host header to override in HTTP request")
-var tooSoon = flag.Int("too-soon", 76, "If NextUpdate is fewer than this many hours in future, warn.")
-var ignoreExpiredCerts = flag.Bool("ignore-expired-certs", false, "If a cert is expired, don't bother requesting OCSP.")
-var expectStatus = flag.Int("expect-status", 0, "Expect response to have this numeric status (0=good, 1=revoked)")
+var (
+	method             = flag.String("method", "GET", "Method to use for fetching OCSP")
+	urlOverride        = flag.String("url", "", "URL of OCSP responder to override")
+	hostOverride       = flag.String("host", "", "Host header to override in HTTP request")
+	tooSoon            = flag.Int("too-soon", 76, "If NextUpdate is fewer than this many hours in future, warn.")
+	ignoreExpiredCerts = flag.Bool("ignore-expired-certs", false, "If a cert is expired, don't bother requesting OCSP.")
+	expectStatus       = flag.Int("expect-status", -1, "Expect response to have this numeric status (0=Good, 1=Revoked, 2=Unknown)")
+	expectReason       = flag.Int("expect-reason", -1, "Expect response to have this numeric revocation reason (0=Unspecified, 1=KeyCompromise, etc.)")
+)
+
+// Config contains fields which control various behaviors of the
+// checker's behavior.
+type Config struct {
+	method             string
+	urlOverride        string
+	hostOverride       string
+	tooSoon            int
+	ignoreExpiredCerts bool
+	expectStatus       int
+	expectReason       int
+}
+
+// DefaultConfig is a Config populated with the same defaults as if command-line
+// flags had been parsed, but none had been passed.
+var DefaultConfig = Config{
+	method:             *method,
+	urlOverride:        *urlOverride,
+	hostOverride:       *hostOverride,
+	tooSoon:            *tooSoon,
+	ignoreExpiredCerts: *ignoreExpiredCerts,
+	expectStatus:       *expectStatus,
+	expectReason:       *expectReason,
+}
+
+var configFromFlags Config
+var isConfigPopulated bool = false
+
+// ConfigFromFlags returns a Config whose values are populated from
+// any command line flags passed by the user, or default values if not passed.
+func ConfigFromFlags() Config {
+	if !isConfigPopulated {
```

Comment:
This would be a little nicer as a [`sync.Once`](https://golang.org/pkg/sync/#Once), which gives protection from race conditions. I don't think we ever call ConfigFromFlags multiple times concurrently, but might as well future-proof.

Reply:
Nice. Yeah, there's no existing situation in which that would happen but no reason to use existing memoization utils.